File: DataStructures/php/Stack/Stack.php
<?php

class Stack
{
    public int $top = -1;

    public function __construct()
    {
        $this->stack = [];
        
    }

    public function push($value)
    {
        $this->top ++;
        
        $this->stack[] = $value;
    }

    public function pop()
    {
        if($this->isEmpty())
        {
            throw new Exception('Stack is empty');
        }

        $top = $this->stack[$this->top];
        unset($this->stack[$this->top]);

        $this->top --;

        return $top;
    }

    public function top()
    {
        return $this->stack[$this->top];
    }

    public function isEmpty()
    {
        return $this->size() <= 0;
    }

    public function size()
    {
        return count($this->stack);
    }
}

$stack = new Stack();
echo 'stack size: ' . $stack->size() . PHP_EOL;

try{
    $stack->pop();
}catch(Exception $e)
{
    echo $e->getMessage() . PHP_EOL;
}

$stack->push(1);
echo 'stack size: ' . $stack->size() . PHP_EOL;
$stack->push(2);
echo 'stack size: ' . $stack->size() . PHP_EOL;
echo 'stack top: ' . $stack->top() . PHP_EOL;
$top = $stack->pop();
echo 'stack pop: ' . $top . PHP_EOL;
echo 'stack top: ' . $stack->top() . PHP_EOL;
echo 'stack size: ' . $stack->size() . PHP_EOL;
echo 'stack pop: ' . $stack->pop() . PHP_EOL;
if($stack->isEmpty()){
    echo "stack is Empty Now" . PHP_EOL;
}

File: DataStructures/python/Stack/Stack.py
from enum import Flag
from mechanize import Item


class Stack:
    def __init__(self) -> None:
        self.stack = [];
        self.top = 0

    def pop(self):
        if self.is_empty():
            return None

        self.top -= 1
        return self.stack.pop()
        
    def push(self, value):
        self.top += 1
        self.stack.append(value)

    def top(self):
        return self.stack[self.top]

    def size(self):
        return len(self.stack)

    def is_empty(self):
        return self.top <= 0

stack = Stack();
assert stack.is_empty() == True
assert stack.pop() == None
assert stack.size() == 0


stack.push(1)
assert stack.is_empty() == False
assert stack.size() == 1
assert stack.pop() == 1
assert stack.size() == 0

stack.push(1)
stack.push(2)
assert stack.is_empty() == False
assert stack.size() == 2
assert stack.pop() == 2
assert stack.size() == 1
assert stack.is_empty() == False
assert stack.pop() == 1
assert stack.size() == 0